src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java

```diff
@@ -128,8 +134,21 @@ public void onTraversalStart() {
         final IntervalListScatterer scatterer = subdivisionMode.make();
         final List<IntervalList> scattered = scatterer.scatter(intervalList, scatterCount);
 
+        // optionally split interval lists that contain intervals from multiple contigs
+        final List<IntervalList> scatteredFinal = !dontMixContigs ? scattered :
+                scattered.stream().flatMap(il -> il.getIntervals().stream()
+                        .collect(Collectors.groupingBy(Interval::getContig)).entrySet().stream()    // group each interval list into sublists
```

Comment:
consider using ctrl + alt + L here, to fix the spacing here

Reply:
done